java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java

```diff
@@ -30,17 +30,21 @@
   static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(BoundsChecking.class);
 
   static {
+    String envProperty = System.getenv("ARROW_ENABLE_UNSAFE_MEMORY_ACCESS");
     String oldProperty = System.getProperty("drill.enable_unsafe_memory_access");
     if (oldProperty != null) {
       logger.warn("\"drill.enable_unsafe_memory_access\" has been renamed to \"arrow.enable_unsafe_memory_access\"");
       logger.warn("\"arrow.enable_unsafe_memory_access\" can be set to: " +
               " true (to not check) or false (to check, default)");
     }
-    boolean isAssertEnabled = false;
-    assert isAssertEnabled = true;
-    BOUNDS_CHECKING_ENABLED = isAssertEnabled ||
-      !"true".equals(System.getProperty("arrow.enable_unsafe_memory_access")) ||
-      !"true".equals(oldProperty);
+    String newProperty = System.getProperty("arrow.enable_unsafe_memory_access");
+
+    // The priority of determining the unsafe flag:
+    // 1. The system properties take precedence over the environmental variable.
+    // 2. The new system property takes precedence over the new system property.
+    String unsafeFlagValue = newProperty != null ? newProperty : (oldProperty != null ? oldProperty : envProperty);
```

Comment:
stacking ternary operators is quite hard to read, suggest against it in general.

Reply:
Thanks for the suggestion. I have replace them with if conditions.